scripts/apps/authoring/authoring/services/AuthoringService.ts

```diff
@@ -368,6 +368,15 @@ export function AuthoringService($q, $location, api, lock, autosave, confirm, pr
         helpers.filterDefaultValues(extDiff, orig);
         var endpoint = 'archive_' + action;
 
+        if (appConfig.features.publishFromPersonal && !orig?.task?.desk && !extDiff?.task?.desk) {
+            var currentDeskId = session.identity.desk || desks.getCurrentDeskId();
```

Comment:
It's not very good that you need a fallback here. It would be good to fix one of the methods to it works reliably.